layers/+lang/emacs-lisp/packages.el

```diff
@@ -46,6 +47,24 @@
 (defun emacs-lisp/post-init-eldoc ()
   (add-hook 'emacs-lisp-mode-hook 'eldoc-mode))
 
+(defun emacs-lisp/init-auto-compile ()
+  (use-package auto-compile
+    :defer t
+    :init
+    (progn
+      (setq auto-compile-display-buffer nil
+            auto-compile-use-mode-line t
+            auto-compile-mode-line-counter t)
+      (add-hook 'emacs-lisp-mode-hook
+                (lambda () (auto-compile-on-save-mode 1)))
+      (add-hook 'lisp-interaction-mode-hook
+                (lambda () (auto-compile-on-save-mode 1))))
```

Comment:
Ah, this is not actually correct.  `auto-compile-on-save-mode` is the _global_ variant.  What we actually need here is just [auto-compile-mode](https://github.com/tarsius/auto-compile/blob/master/auto-compile.el#L127):

```
(add-hook 'emacs-lisp-mode-hook 'auto-compile-mode)
```

Besides, I don't think that we need compilation for `lisp-interaction-mode`…